contracts/oracle/VoltSystemOracle.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {Constants} from "./../Constants.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import {IVoltSystemOracle} from "./IVoltSystemOracle.sol";
+
+/// @notice contract that receives a fixed interest rate upon construction,
+/// and then linearly interpolates that rate over a 1 year period into the VOLT price
+/// after the oracle start time.
+/// Interest can compound annually. Assumption is that this oracle will only be used until
+/// Volt 2.0 ships. Maximum amount of compounding periods on this contract at 2% APR
+/// is 6192 years, which is more than enough for this use case.
+/// @author Elliot Friedman
+contract VoltSystemOracle is IVoltSystemOracle {
+    /// ---------- Mutable Variables ----------
+
+    /// @notice acts as an accumulator for checkpointing interest earned in previous periods
+    uint256 public override oraclePrice;
+
+    /// @notice start time at which point interest will start accruing, and the
+    /// point in time at which the current ScalingPriceOracle price will be
+    /// snapshotted and saved
+    uint256 public override oracleStartTime;
```

Comment:
Nit, this isn't really an `oracleStartTime` but a `periodStartTime` or similar, as you update it when interest compounds

Reply:
updated!